index.bs

```diff
@@ -3820,6 +3820,7 @@ partial interface MLGraphBuilder {
     The <dfn method for=MLGraphBuilder>layerNormalization(|input|, |options|)</dfn> method steps are:
   </summary>
   <div class=algorithm-steps>
+    1. If |options|.{{MLLayerNormalizationOptions/axes}} does not [=map/exist=], set |options|.{{MLLayerNormalizationOptions/axes}} to the [=/list=] « 1, 2, 3 ».
```

Comment:
Hmm, the spec wording doesn't quite match the intention nor the Chromium implementation, as that example of [1,2,3] is only accurate for *4D* input tensors. The intention is that if not specified, axes defaults to [1,2,...N-1], and so for 3D, effective axes = [1,2].

Chromium: https://chromium-review.googlesource.com/c/chromium/src/+/5068275
```
    if (input_rank > 1) {
      // Default to `input_dimensions[1, ... , N-1]` where N is the input rank.
      reduction_dimensions.assign(input_dimensions.begin() + 1,
                                  input_dimensions.end());
```

WPT: https://github.com/webmachinelearning/webnn-baseline/pull/60/files#diff-0f9a981097ee5210d3ed917824f3a9fe72917311e38bc99d7972ea9009634f9dR32
```
  if (axes === undefined) {
    axes = Array.from({length: input.rank - 1}, (_, index) => index + 1);
  }
```

Reply:
The Infra [exclusive range](https://infra.spec.whatwg.org/#the-exclusive-range) for that. So naively it'd be:

```
1. If |options|.{{MLLayerNormalizationOptions/axes}} does not [=map/exist=], then set |options|.{{MLLayerNormalizationOptions/axes}} to a new [=/list=] equal to [=the range=] 1 to |input|'s [=rank-=], exclusive. 
```

However... what if _input_'s rank is 0 (scalar) or 1 ? Empty list in both cases?

Meta comment: although we're slowly evolving "standardese" into a more formal language, it's a long long road. The important thing is that it is as unambiguous as we can make it, and doesn't assume domain knowledge... because engineers are creative! (Also that it has tests.) So don't feel beholden to specific spec syntax, although if it exists we should use it!

Reply:
> what if input's rank is 0 (scalar) or 1 ? Empty list in both cases?

Correct.

Reply:
If empty list then I'd probably write:
```
1. If |options|.{{MLLayerNormalizationOptions/axes}} does not [=map/exist=], then set |options|.{{MLLayerNormalizationOptions/axes}} to a new [=/list=] equal to [=the range=] 1 to |input|'s [=rank-=], exclusive, if |input|'s [=rank=] is greater than 1, and an empty [=/list=] otherwise.
```
... or break it down into substeps, but I think that's overkill.

Reply:
Substeps seems overkill. How about adding "from" and either "rank - 1" or "end exclusive". e.g.

`...equal to [=a range=] from 1 to |input|'s [=rank-=] - 1, if |input|'s [=rank=] is greater than 1, ..."

Reply:
Editor's call. :) Infra suggests "The range n to m, exclusive" https://infra.spec.whatwg.org/#the-exclusive-range or "The range n to m, inclusive" https://infra.spec.whatwg.org/#the-exclusive-range - without "from" and the inclusive/exclusive applies to the end implicitly, but whatever makes it clearer for readers.

(FYI: Infra says a range is an "ordered set", but that's still just a list.)

Reply:
BTW, with Write permissions should be able to make edits to the PR directly on GitHub. Click on the **Files changed** tab, click on the "···" menu for the `index.bs` file, click **Edit**, make a change, click **Commit changes...**. Please do so if it will save some round-trips!

Reply:
⏳ Reading the links you supplied (thanks).

> Infra suggests ... without "from"
> set options.axes (to) a new list equal (to) the range 1 (to) input's rank

Yeah, normally I'd be fine without the "from", but the series of 3 `to`'s in a row just made it harder mentally parse 😅.

> Edit, make a change, click Commit changes

Alrighty, if you're comfortable with that (I usually don't stomp over other's CR's). There's also the "```suggestion" boxes which are really convenient for both parties and allow some negotiation rather than a direct edit. I'll add one now...